### PR TITLE
Add TensorRT timing cache feature

### PR DESCRIPTION
**Pull request by chilo-ms**
TensorRT provides the timing cache feature to reduce the builder time by keeping the layer profiling information during the builder phase. This PR added timing cache feature into ORT-TRT.

Also, please notice that ORT won't use `OrtTensorRTProviderOptions` struct anymore for TRT EP when adding additional provider option. Instead, it uses the opaque struct `OrtTensorRTProviderOptionsV2` as internal struct for setting provider options that can be converted to a string.
Please see https://github.com/microsoft/onnxruntime/pull/7808 and https://github.com/microsoft/onnxruntime/pull/10188 for more details and context.

### CODE REVIEW

**Comment by chilo-ms** [comment]
Need to add unit test for testing this feature.

File: onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc [hunk 1]
```diff
@@ -1302,6 +1338,7 @@ common::Status TensorrtExecutionProvider::Compile(const std::vector<Node*>& fuse
     if (!has_dynamic_shape) {
       const std::string cache_path = GetCachePath(cache_path_, trt_node_name_with_precision);
       const std::string engine_cache_path = cache_path + ".engine";
+      const std::string timing_cache_path = cache_path + ".timing";
```
Comment (jywu-msft) [inline onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc:1341]: i'm wondering if we should put the timing cache in the same location as engine cache/profiles 
the engines may be model specific, but timing cache can be system wide?

**Comment by jywu-msft** [comment]
> Will add unit test for testing this feature.

yes, we definitely need some test cases here.
need to test in conjunction with engine cache enabled/disabled as well.

**Comment by chilo-ms** [comment]
> > Will add unit test for testing this feature.
> 
> yes, we definitely need some test cases here. need to test in conjunction with engine cache enabled/disabled as well.

Test cases for timing cache have been added

**Comment by gedoensmax** [comment]
I think we can close this dur to #14767 right ?

**Comment by chilo-ms** [comment]
> I think we can close this dur to #14767 right ?

yes, we can

**Pull request closed without merging**